plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -129,7 +131,7 @@ public CachingJdbcClient(
         tableHandlesByQueryCache = buildCache(cacheMaximumSize, metadataCachingTtl);
         procedureHandlesByQueryCache = buildCache(cacheMaximumSize, metadataCachingTtl);
         columnsCache = buildCache(cacheMaximumSize, metadataCachingTtl);
-        statisticsCache = buildCache(cacheMaximumSize, metadataCachingTtl);
+        statisticsCache = buildCache(cacheMaximumSize, statisticsCachingTtl);
```

Comment:
If `metadataCachingTtl` is set higher than `statisticsCachingTtl`, then that should be used as the value for `statisticsCachingTtl` (we did that in hive metadata caching to avoid inadvertently using smaller ttl for stats cache than overall metadata cache).

Reply:
if we want such logic, this should rather be a validation, not silently ignored config

Reply:
If we were to ever provide some default value for statisticsCachingTtl, then there's a chance that this default could be lower than the existing value of metadataCachingTtl setup by the user. In this case, using smaller value of statisticsCachingTtl would potentially regress perf and applying validation would force user to read docs and update the config values. Applying validation strictly is a correct but disruptive choice, we chose to do the right thing silently in the case of hive.
cc: @sopel39

Reply:
In hive we just use
```
        Duration metastoreCacheTtl = config.getMetastoreCacheTtl();
        Duration statsCacheTtl = config.getStatsCacheTtl();
        if (metastoreCacheTtl.compareTo(statsCacheTtl) > 0) {
            statsCacheTtl = metastoreCacheTtl;
        }
```
it mitigates all bad configs and I think it's OK (less troubles for us)

Reply:
https://github.com/trinodb/trino/pull/19862